ext/pcntl/pcntl.c

```diff
@@ -986,7 +991,7 @@ PHP_FUNCTION(pcntl_signal)
 			php_error_docref(NULL, E_WARNING, "Error assigning signal");
 			RETURN_FALSE;
 		}
-		zend_hash_index_del(&PCNTL_G(php_signal_table), signo);
+		zend_hash_index_update(&PCNTL_G(php_signal_table), signo, handle);
```

Comment:
Well, zend_call_function() will properly make sure that we're dealing with a callable and fail gracefully.  That said, it's simple (already added) a check to make sure the handle isn't a long.